src/datasources/accounts/accounts.datasource.spec.ts

```diff
@@ -68,21 +102,59 @@ describe('AccountsDatasource tests', () => {
 
       const result = await target.getAccount(address);
 
-      expect(result).toStrictEqual({
-        id: expect.any(Number),
-        group_id: null,
-        address,
-        created_at: expect.any(Date),
-        updated_at: expect.any(Date),
-      });
+      expect(result).toStrictEqual(
+        expect.objectContaining({
+          id: expect.any(Number),
+          group_id: null,
+          address,
+        }),
+      );
     });
 
-    it('throws if no account is found', async () => {
+    it('returns an account from cache', async () => {
       const address = getAddress(faker.finance.ethereumAddress());
+      await target.createAccount(address);
+
+      // cache the account
+      await target.getAccount(address);
+
+      // delete the account from the database
+      await sql`DELETE FROM accounts WHERE address = ${address}`;
+
+      // check the account is still in the cache
+      const result = await target.getAccount(address);
```

Comment:
If the account has been deleted, we shouldn't keep it in cache.

Reply:
It's deleted from the database using a raw SQL query, so it is still present in the cache. Maybe this is a bit weird way to test the cache contents, in other tests the cache contents are read and checked directly using `fakeCacheService.get`, I combined both approaches in the tests, and this is a more _complete_ one because it gets the content using `getAccount(address)`, but maybe reading the cache directly is enough. Will change this approach 👍🏻

Reply:
The testing approach is changed in order to check the cache contents directly in [c611031](https://github.com/safe-global/safe-client-gateway/pull/1760/commits/c6110318a438e3d278ca59a05fc49f9b20ace5d6)